openverse_catalog/dags/maintenance/add_license_url.py

```diff
@@ -0,0 +1,179 @@
+    postgres_conn_id: str,
+) -> Literal["update_license_url", "final_report"]:
+    logger.info("Getting image records without license_url in meta_data.")
+    postgres = PostgresHook(postgres_conn_id=postgres_conn_id)
+    null_meta_data_count = postgres.get_first(
+        dedent(
+            """
+        SELECT COUNT(*) from image
+        WHERE meta_data IS NULL;
+        """
+        )
+    )[0]
+
+    logger.info(f"There are {null_meta_data_count} records with NULL in meta_data.")
+    if null_meta_data_count > 0:
+        return "update_license_url"
+    else:
+        return "final_report"
+
+
+def update_license_url(postgres_conn_id: str) -> dict[str, int]:
+    """Add license_url to meta_data batching all records with the same license.
+    :param postgres_conn_id: Postgres connection id
+    """
+
+    logger.info("Getting image records without license_url.")
+    postgres = PostgresHook(postgres_conn_id=postgres_conn_id)
+
+    total_count = 0
+    total_counts = {}
+    for license_items, path in license_map.items():
+        license_name, license_version = license_items
+        logger.info(f"Processing {license_name} {license_version}, {license_items}.")
+        license_url = f"{base_url}{path}/"
+
+        select_query = dedent(
+            f"""
+            SELECT identifier FROM image
+            WHERE (
+            meta_data is NULL AND license = '{license_name}'
+            AND license_version = '{license_version}')
+            """
+        )
+        result = postgres.get_records(select_query)
+
+        if not result:
+            logger.info(f"No records to update with {license_url}.")
+            continue
+        logger.info(f"{len(result)} records to update with {license_url}.")
+        license_url_col = {"license_url": license_url}
+        update_license_url_query = dedent(
+            f"""
+            UPDATE image
+            SET meta_data = {Json(license_url_col)}
+            WHERE identifier IN ({','.join([f"'{r[0]}'" for r in result])});
+            """
+        )
+
+        updated_count = postgres.run(
+            update_license_url_query, autocommit=True, handler=RETURN_ROW_COUNT
+        )
+        logger.info(f"{updated_count} records updated with {license_url}.")
+        total_counts[license_url] = updated_count
+        total_count += updated_count
+
+    logger.info(f"{total_count} image records with missing license_url updated.")
+    logger.info(f"Total updated counts: {total_counts}")
+    return total_counts
+
+
+def final_report(postgres_conn_id: str, item_count):
+    logger.info(
+        "Added license_url to all items. Checking for any records "
+        "that still have NULL metadata."
+    )
+    postgres = PostgresHook(postgres_conn_id=postgres_conn_id)
+    null_meta_data_records = postgres.get_first(
+        dedent("SELECT COUNT(*) from image WHERE meta_data IS NULL;")
+    )[0]
+    logger.info(f"There are {null_meta_data_records} records with NULL meta_data.")
```

Comment:
This is the same logic as the function above I think, can we combine it into a utility function that both tasks call?

Reply:
Great idea, thank you!